reactioncommerce/reaction
# (refactor): add isBackorder data to variants

Resolves #4814   
Impact: **minor**  
Type: **refactor**

## Issue
The `isBackorder` property is only added to a top-level Catalog item and not added to the item's variants or options.

## Solution
- Add isBackorder field to the schemas for `Catalog` and `Product`
- Add `isBackorder` to Graphql endpoint for `Catalog`
- Add `isBackorder` computation in `createCatalogProduct` [xform](https://github.com/reactioncommerce/reaction/blob/b49fbf03519fecd58925929a01ec17475d6d2e0c/imports/plugins/core/catalog/server/no-meteor/utils/createCatalogProduct.js#L131)

## Breaking changes
None

## Testing
1. Publish a product to the catalog.
1. See that `isBackorder` field is present in Mongo
1. See that `isBackorder` field is present in Catalog Graphql endpoint
